### ISSUE TITLE
flang: ICE 'ILM file can not find intrinsic .csqrt' compiling Octopus jellium.F90

### ISSUE THREAD

**Issue by swanwu1996** [llvm/llvm-project#199331]
## Description

flang (AMD AOCC 5.1.0, based on LLVM flang 17.0.6) crashes with an internal compiler error when compiling the file `src/species/jellium.F90` from [Octopus](https://gitlab.com/octopus-code/octopus) 16.4. The ICE is triggered by a `complex(real64) function` with a `cmplx()` call, where the frontend internally generates a `csqrt` intrinsic that does not exist in the ILM library.

## Error

```
F90-F-0000-Internal compiler error. Errors in ILM file 1 (jellium.F90-pp.f90)
ILM file: can't find intrinsic .csqrt
```

## Environment

- CPU: AMD EPYC 9V74 (Zen4)
- OS: Rocky Linux 9
- AOCC 5.1.0 (flang 17.0.6, CLANG: AOCC 5.1.0-Build 1994 2025 12 23)

## How to reproduce

Extracted and minimized from Octopus 16.4 `src/species/jellium.F90`:

```fortran
module test_bug_m
  use, intrinsic :: iso_fortran_env, only: real64
  implicit none
  type :: species_user_defined_t
  contains
    procedure :: user_pot => jellium_userdef_pot
  end type species_user_defined_t
contains
  complex(real64) function jellium_userdef_pot(spec, dim, xx, r)
    class(species_user_defined_t), intent(in) :: spec
    integer, intent(in) :: dim
    real(real64), intent(in) :: xx(:)
    real(real64), intent(in) :: r
    real(real64) :: pot_re, pot_im
    pot_re = 0.0_real64
    pot_im = 0.0_real64
    jellium_userdef_pot = cmplx(pot_re, pot_im, real64)
  end function jellium_userdef_pot
end module test_bug_m
```

```bash
flang -O2 -c test_bug.F90
# F90-F-0000-Internal compiler error. Errors in ILM file 1 (csqrt)
```

## What fails to fix the ICE

All of the following still trigger the same `csqrt` ICE:
- All optimization levels: `-O0`, `-O1`, `-O2`, `-O3`
- Changing to `cmplx(pot_re, pot_im, kind=real64)`
- Changing to `complex(pot_re, pot_im, real64)` constructor
- Changing to explicit `%re` / `%im` component assignment
- Changing to `function ... result(res)` with separate `complex(real64) :: res`
- Various `-M` flags: `-Mnovect`, `-Mstandard`, `-Kieee`

## Workaround

Rewriting the function body to not use any complex expression:

```fortran
  complex(real64) function jellium_userdef_pot(spec, dim, xx, r)
    ...
    jellium_userdef_pot = (0.0_real64, 0.0_real64)
  end function
```

This compiles successfully.

## Compilers affected / not affected

| Compiler | Status |
|----------|--------|
| AOCC 5.1.0 (flang) | ICE |
| Intel ifx 2025.1-2025.3 | OK |
| GNU gfortran 10+ | OK |

## Analysis

The `csqrt` call is NOT in the original Fortran source. The Fortran frontend lowers `complex(real64)` expressions by internally creating temporary complex variables and generating a `csqrt` intrinsic call. The ILM runtime library does not provide this intrinsic, causing the ICE.

This blocks compiling Octopus 16.4 (a widely-used real-space TDDFT code) with flang/AOCC on AMD Zen4 systems.

**Comment by eugeneepshteyn** [comment]
@swanwu1996 , this repository is for new LLVM flang. To post issues for classic flang, please use  https://github.com/flang-compiler/flang

I'm going to close this issue.
